vispy/visuals/text/text.py

```diff
@@ -474,7 +477,17 @@ def color(self):
 
     @color.setter
     def color(self, color):
-        self._color = Color(color)
+        if isinstance(color, str):
+            self._color = ColorArray(Color(color))
+        else:
+            color = np.atleast_2d(color).astype(np.float32)
+            if color.shape[1] == 3:
+                color = np.concatenate((color, np.ones((color.shape[0], 1),
```

Comment:
Doesn't `ColorArray` handle this case? If not, should it?

Reply:
Judging by this https://github.com/vispy/vispy/blob/master/vispy/color/color_array.py#L91-L97, this chunk of code shouldn't be needed. Or am I missing what this is doing?

Reply:
no, you are correct, edited in f372f1e8848dd1eae44cf1d6e189b26c8caf1d26

Reply:
Thanks. I added another question about this handling in a separate comment.